### PR TITLE
New component: `discovery.nomad`

### PR DESCRIPTION
**Pull request by captncraig**
Fixes [#2276](https://github.com/grafana/agent/issues/2276)

### CODE REVIEW

File: docs/sources/flow/reference/components/discovery.nomad.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+`tag_separator` | `string` | Seperator to join nomad tags into prometheus labels. | `,` | no
+
+`refresh_interval` | `duration` | Frequency to refresh list of containers. | `"30s"` | no
+`bearer_token` | `secret` | Bearer token to authenticate with. | | no
+`bearer_token_file` | `string` | File containing a bearer token to authenticate with. | | no
+`proxy_url` | `string` | HTTP proxy to proxy requests through. | | no
+`follow_redirects` | `bool` | Whether redirects returned by the server should be followed. | `true` | no
+`enable_http2` | `bool` | Whether HTTP2 is supported for requests. | `true` | no
+
+ At most one of the following can be provided:
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/discovery.nomad.md:37]: Are these optional? Can you skip this? When it says `can be provided`, it implies it's optional. So checking to make sure this is intended to be optional.

```suggestion
 You can provide one of the following arguments for authentication:
```

Or if it's mandatory:

> You must supply one of the following arguments for authentication:

**Review by clayton-cornell — COMMENTED** [review]
Minor comments on docs

**Review by clayton-cornell — APPROVED** [review]
Docs LGTM

File: docs/sources/flow/reference/components/discovery.nomad.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`server` | `string` | Address of nomad server. | `http://localhost:4646` | no
+`namespace` | `string` | Nomad namespace to use. | `default` | no
+`region` | `string` | Nomad region to use. | `global` | no
+`allow_stale` | `bool` | Allow reading from non-leader nomad instances. | `true` | no
+`tag_separator` | `string` | Seperator to join nomad tags into Prometheus labels. | `,` | no
+
```
Comment (marctc) [inline docs/sources/flow/reference/components/discovery.nomad.md:29]: this newline breaks the table

File: docs/sources/flow/reference/components/discovery.nomad.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+The following fields are exported and can be referenced by other components:
+
+Name | Type | Description
+---- | ---- | -----------
+`targets` | `list(map(string))` | The set of targets discovered from the nomad server.
+
+Each target includes the following labels:
+
+TODO:
```
Comment (marctc) [inline docs/sources/flow/reference/components/discovery.nomad.md:93]: is this TODO intended?

File: component/discovery/nomad/nomad_test.go [hunk 4]
```diff
@@ -0,0 +1,57 @@
+package nomad
```
Comment (marctc) [inline component/discovery/nomad/nomad_test.go:1]: Should we add a test for validate?

**Review by marctc — APPROVED** [review]

**Pull request merged**